solidity/contracts/cross-chain/wormhole/L1BTCDepositorNttWithExecutor.sol

```diff
@@ -0,0 +1,1102 @@
+            "Recipient address cannot be zero when platform fee is set"
+        );
+        address oldRecipient = defaultPlatformFeeRecipient;
+        defaultPlatformFeeRecipient = _newRecipient;
+        emit DefaultPlatformFeeRecipientUpdated(oldRecipient, _newRecipient);
+    }
+
+    /// @notice Updates the underlying NTT Manager address
+    /// @param _newNttManager New underlying NTT Manager address
+    function updateUnderlyingNttManager(address _newNttManager)
+        external
+        onlyOwner
+    {
+        require(
+            _newNttManager != address(0),
+            "NTT Manager address cannot be zero"
+        );
+        address oldManager = underlyingNttManager;
+        underlyingNttManager = _newNttManager;
+        emit UnderlyingNttManagerUpdated(oldManager, _newNttManager);
+    }
+
+    /// @notice Updates the NTT Manager With Executor address
+    /// @param _newNttManagerWithExecutor New NTT Manager With Executor address
+    function updateNttManagerWithExecutor(address _newNttManagerWithExecutor)
+        external
+        onlyOwner
+    {
+        require(
+            _newNttManagerWithExecutor != address(0),
+            "Address cannot be zero"
+        );
+        address oldManager = address(nttManagerWithExecutor);
+        nttManagerWithExecutor = INttManagerWithExecutor(
+            _newNttManagerWithExecutor
+        );
+        emit NttManagerWithExecutorUpdated(
+            oldManager,
+            _newNttManagerWithExecutor
+        );
+    }
+
+    /// @notice Allows the owner to retrieve tokens from the contract
+    /// @param _token The address of the token to retrieve (address(0) for native token)
+    /// @param _to The address to send the tokens to
+    /// @param _amount The amount of tokens to retrieve
+    function retrieveTokens(
+        address _token,
+        address _to,
+        uint256 _amount
+    ) external onlyOwner {
+        require(
+            _to != address(0),
+            "Cannot retrieve tokens to the zero address"
+        );
+
+        if (_token == address(0)) {
+            // Use call instead of transfer for better error handling and gas efficiency
+            // solhint-disable-next-line avoid-low-level-calls
+            (bool success, ) = payable(_to).call{value: _amount}("");
+            require(success, "Failed to transfer native token");
+        } else {
+            IERC20Upgradeable(_token).safeTransfer(_to, _amount);
+        }
+    }
+
+    /// @notice Sets executor parameters and returns the nonce for reference
+    /// @param executorArgs Real executor arguments with valid signed quote from Wormhole Executor API
+    /// @param feeArgs Fee arguments for the executor service
+    /// @return nonce The nonce hash for these parameters (for informational purposes)
+    /// @dev Must be called before finalizeDeposit() to provide real signed quote
+    function setExecutorParameters(
```

Comment:
Is staging executor parameters necessary? Does it make sense to consider allowing callers to pass ExecutorArgs/FeeArgs directly in a dedicated method (calculated off-chain), removing nonce/expiry/storage overhead for the common path?

Reply:
we discusses this yesterday, the `setParameter()` is part of the `finalizedDeposit()` segment, just keep it separately to keep `finalizeDeposit()` signature. You suggested doing a `MultiCall()` to have atomic call of both.